### ISSUE TITLE
AppendiXNet: Deep Learning for  Diagnosis of Appendicitis from A  Small Dataset of CT Exams Using  Video Pretraining

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#51]
## Abstract & Intro.
* 3D deep learning model인 AppendiXNet를 제안함.
* a deep learning model capable of detecting appendicitis(맹장염) on abdominal CT using a small dataset of 438 CT examinations
* 모델 학습 흐름
> 1. AppendiXNet was pretrained on a large collection of YouTube videos called Kinetics, 
consisting of approximately 500,000 video clips and annotated for one of 600 human action classes
> 2. fine-tuned on a small dataset of 438 CT scans annotated for appendicitis
* pretraining the 3D model on natural videos signifcantly improved the performance of the model from an AUC of 0.724 (95% CI 0.625, 0.823) to 0.810 (95% CI 0.725, 0.895)
* -> 자연 이미지로 pre-training하면 medical imaging tasks의 정확도도 높일 수 있다

### Data
* CT exams of the abdomen and pelvis (하나의 GE Medical Systems scanner에서 얻음.)
* 3개의 sets로 분리
> 1. a training set used to train the model
> 2. a development set used for model selection
> 3. a test set to evaluate the final performance of the model
* test set의 ground truth label 분류
  * appendicitis 라벨을 붙이는 조건
  > (a) the patient’s radiology report at the time of exam was rendered a diagnosis of acute appendicitis
  > (b) appendicitis was confirmed upon manual review of the CT scan by the aforementioned radiologist
  > (c) the patient underwent appendectomy and inflammation of the appendix was confirmed by the pathology report
  * negative 라벨을 붙이는 조건
  > (a) the patient was scanned for abdominal pain and a clinical concern for appendicitis
  > (b) the radiology report at the time of exam did not render a diagnosis of acute appendicitis
  > (c) a normal appendix was confirmed upon manual review of the CT scan by the aforementioned radiologist
  > (d) the patient did not undergo appendectomy or image-guided percutaneous drainage as a result of the index CT and was discharged or treated for another unrelated diagnosis (e.g. renal stone)

* Data set selection flow
<image src="https://user-images.githubusercontent.com/67627471/129034076-6ac77934-ca0a-47d8-aa66-9e3f4aaebf6c.png" width="80%" height="80%" />

### Model development
* AppendiXNet
  > 18-layer 3D CNN for classification of appendicitis from CT exams
  > takes in groups of 8-slices and produces a single probability for the group
* When there are not enough slices to fill the group, the group is padded with slices containing all zeros.
* The probabilities for all of the groups within a scan are combined by first applying a rolling mean with a window size of 3 to the group probabilities, and then computing the maximum average probability over the resulting means as the probability of appendicitis in the entire scan.
1. AppendiXNet was first pretrained on a large collection of YouTube videos called Kinetics. ( 대략 500,000개의 video clips이고 각각 10초 정도의 길이임.)
2. 비디오 클립은 600개의 action classes로 annotate됨.
3. pretraining 후에 마지막 fc (fully connected) layer를 제거하고, 맹장염 task로 fine-tuning시킴.
4. AppendiXNet의 성능을 pretrainig 시킨 것과 안 시킨 것으로 나눠 성능을 비교함.

* pre-training과 fine-tuning 과정
<image src="https://user-images.githubusercontent.com/67627471/129036267-5d6a3565-042d-446e-a83d-0eea39a11360.png" width="80%" height="80%" />

### Model architecture
* use a 3D ResNet architecture (7x7x7 conv layer + 3x3x3 max pooling layer + 4 stacked residual blocks [=>3x3x3 conv layer, 3D batch norm layer, ReLU nonlinearity, 3x3x3 conv layer, downsampling layer, residual connection, final ReLU nonlinearily] + global average pooling + fc layer[single output] + sigmoid nonlinearity[final probability 내놓음] ) (그림 하나 그려주지 이해하기 힘들게 참..)

## Result
* pretraining 시킨 모델과 안 시킨 모델의 성능 비교
![image](https://user-images.githubusercontent.com/67627471/129038961-032b0c93-7281-4c72-811c-5480af4b886c.png)

* AppendiXNet의 prediction을 Grad-CAM으로 interpreting해서 봄.
![image](https://user-images.githubusercontent.com/67627471/129041054-4fc33ad7-ccde-45c9-94dc-86b65d125199.png)

* appendiXNet말고도 다른 모델을 pretraining시킨 것/안 시킨 것으로 나누어 비교해봄.
<image src="https://user-images.githubusercontent.com/67627471/129041175-5c9eac8c-b4c5-42a2-882f-e7622b812c2c.png" width="50%" height="50%" />

##
* kinetic 영상으로 pre-training한 것이 맹장염 classification 성능을 높인다니.. 심지어 영상과 medical image는 domain이 완전 다르지 않나
* small dataset은 medical dataset의 어쩔 수 없는 한계인데, 이미지 외의 medical dataset에도 이런 연구가 많이 이루어졌으면 함.
